tensorflow/python/data/kernel_tests/matching_files_dataset_op_test.py

```diff
@@ -0,0 +1,267 @@
+    filenames = ['a.txt', 'b.py', 'c.pyc']
+    self._touchTempFiles(filenames)
+
+    filename_placeholder = array_ops.placeholder(dtypes.string, shape=[])
+    dataset = MatchingFilesDataset(filename_placeholder)
+
+    with self.cached_session() as sess:
+      itr = iterator_ops.Iterator.from_structure(dataset.output_types)
+      init_op = itr.make_initializer(dataset)
+      next_element = itr.get_next()
+      sess.run(
+        init_op,
+        feed_dict={filename_placeholder: path.join(self.tmp_dir, '*.py*')})
+
+      full_filenames = []
+      produced_filenames = []
+      for filename in filenames[1:]:
+        full_filenames.append(
+          compat.as_bytes(path.join(self.tmp_dir, filename)))
+        produced_filenames.append(compat.as_bytes(sess.run(next_element)))
+
+      self.assertItemsEqual(full_filenames, produced_filenames)
+
+      with self.assertRaises(errors.OutOfRangeError):
+        sess.run(itr.get_next())
+
+  def _load_data(self):
+    new_files = []
+    dir = "/tmp/test/"
+    if not os.path.exists(dir):
+      os.makedirs(dir)
+    base = tempfile.mkdtemp(prefix=dir)
+    print('saving files to dir: {}'.format(base))
+    for i in range(width):
+      new_base = os.path.join(base, str(i), *[str(j) for j in range(depth - 1)])
+      if not os.path.exists(new_base):
+        os.makedirs(new_base)
+      f = os.path.join(new_base, 'stuff.txt')
+      new_files.append(compat.as_bytes(f))
+      open(f, 'w').close()
+    return base, new_files
+
+  def _read_data(self, data, sess, N=1):
+    for _ in range(N):
+      sess.run(data)
+
+  def _read_data_with_result(self, data, sess, N=1):
+    result = []
+    for _ in range(N):
+      result.append(sess.run(data))
+    return result
+
+  def testPerformance(self):
+    base, test_filenames = self._load_data()
+    test_filenames.sort(reverse=True)
+    patterns = array_ops.placeholder(dtypes.string, shape=[None])
+    dataset = MatchingFilesDataset(patterns)
+    iterator = iterator_ops.Iterator.from_structure(dataset.output_types)
+    init_op = iterator.make_initializer(dataset)
+    get_next = iterator.get_next()
+    result = []
+    with self.cached_session() as sess:
+      pattern = '{}/{}/*.txt'\
+        .format(base, os.path.join(*['**' for _ in range(depth)]))
+      search_patterns = [pattern]
+      sess.run(init_op, feed_dict={patterns: search_patterns})
+      result.extend(timeit(partial(self._read_data_with_result, get_next, sess),
+        "read first filename"))
+      result.extend(timeit(partial(self._read_data_with_result, get_next, sess),
+        "read second filename"))
+      N = width * len(search_patterns) - 2
+      filename = timeit(partial(self._read_data_with_result, get_next, sess, N),
+        'read {} more filenames'.format(N), N)
+      result.extend(filename)
+
+    matched_filenames = [compat.as_bytes(x) for x in result]
+    self.assertEqual(matched_filenames, test_filenames)
+
+  def testShuffle(self):
```

Comment:
What is this testing that `testPerformance` is not? The fact that shuffling work correctly? I don't see how that's related to the logic introduced in this CL.

Reply:
This is not related to the logic introduced in this CL. Have removed this testing.